agroal-pool/src/main/java/io/agroal/pool/Poolless.java

```diff
@@ -313,6 +317,31 @@ public long awaitingCount() {
         return synchronizer.getQueueLength();
     }
 
+    // --- health check //
+
+    @Override
+    public boolean isHealthy(boolean connect) throws SQLException {
+        // the main difference here is that connect will not block (and potentially timeout) on full pool
+        if ( connect ) {
```

Comment:
You can't go over max-pool-size as that setting may be enforced by another layer or the database itself

Reply:
I can go over max! it's a matter for who dimensions the system.

That's why there are two modes and only one of them causes the size to go over max.